packages/transaction-controller/src/helpers/PendingTransactionTracker.ts

```diff
@@ -232,7 +251,14 @@ export class PendingTransactionTracker {
   async #checkTransactions() {
     this.#log('Checking transactions');
 
-    const pendingTransactions = this.#getPendingTransactions();
+    const pendingTransactions: TransactionMeta[] = [
+      ...new Set(
+        [
+          ...this.#getPendingTransactions(),
+          this.#transactionToForcePoll,
+        ].filter((tx): tx is TransactionMeta => tx !== undefined),
```

Comment:
Minor, could we do a `...(this.#transactionToForcePoll ? [this.#transactionToForcePoll] : [])` instead of the `filter`?

Reply:
Applied the changes